### ISSUE TITLE
Backward with custom gradient

### ISSUE THREAD

**Issue by allanj** [clab/dynet#1421]
Can I assign customized gradient to the loss?

For example, currently, we are using the `loss` expression to do backward.
`loss.backward()`

But can I use my own gradient to do backward, let's say my loss is defined somewhere esle and I have my own gradients for the expression and want to use that to do backward.

**Comment by neubig** [comment]
There's not a direct way to do this, but it's pretty easy to do indirectly.

    x_expr = my_neural_net()
    g = gradient_of_x_in_numpy()
    g_expr = dy.inputTensor(g)
    loss = dy.dot_product(x_expr, g_expr)
    loss.forward()
    loss.backward()

What we're doing here is taking advantage of the fact that by taking the dot product, the gradient of `x_expr` will become `g_expr`.

**Comment by neubig** [comment]
Call `dy.sum_elems()` to turn the matrix into a scalar.
